src/adapters/address-adapter.ts

```diff
@@ -0,0 +1,62 @@
+// Declare Address type for object returned from backend API
+type Address = {
+    address: string,
+    latitude: number,
+    longitude: number,
+}
+
+/**
+ * This function will return the best match for given address with coordinates.
+ * @param {String} address - address to search.
+ * @example
+ * const address = "315 Kildaire Farm Rd, Cary, NC 27511";
+ * getAddressCoordinate(address);
+ * @returns {Promise<void | Address>} A promise that is top matching result for given address with coordinates..
+ */
+export const getAddressCoordinate = (address: string): Promise<void | Address> => 
+    // Call backend API to search for given address. ${process.env.HOSTNAME} variable is used to avoid hardcoding hostname
+    // to allow running it locally or remotly without code changes.
+    // Value for HOSTNAME is provided in .env file
+    fetch(`https://{process.env.HOSTNAME}/api/v1/address/search?address=${address}`, {
+        method: 'get',
+    })
```

Comment:
For this fetch, and the one below, import and use fetchAPIData like the other adapter files do. You will also be able to get rid of resp.json.

Reply:
done